hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -331,6 +343,159 @@ newtype ShrinkCount =
   ShrinkCount Int
   deriving (Eq, Ord, Show, Num, Enum, Real, Integral)
 
+-- | Where to start running a property's tests.
+--
+data Skip =
+  -- | Don't skip anything.
+  --
+    SkipNothing
+
+  -- | Skip to a specific test number. If it fails, shrink as normal. If it
+  --   passes, move on to the next test. Coverage checks are disabled.
+  --
+  | SkipToTest TestCount
+
+  -- | Skip to a specific test number and shrink state. If it fails, stop
+  --   without shrinking further. If it passes, the property will pass without
+  --   running any more tests.
+  --
+  --   Due to implementation details, all intermediate shrink states - those on
+  --   the direct path from the original test input to the target state - will
+  --   be tested too, and their results discarded.
+  --
+  | SkipToShrink TestCount ShrinkPath
+  deriving (Eq, Ord, Show, Lift)
+
+-- | We use this instance to support usage like
+--
+-- @
+--   withTests "3:aB"
+-- @
+--
+--   It throws an error if the input is not a valid compressed 'Skip'.
+--
+instance IsString Skip where
+  fromString s =
+    case skipDecompress s of
+      Nothing ->
+        error $ "Not a valid Skip: " ++ s
+      Just skip ->
+        skip
```

Comment:
We use this instance to suggest `withSkip "3:aBc"` to the user for `recheck`. It's maybe a bit overkill? And if the user doesn't have `OverloadedStrings` they're out of luck.

I don't want to add anything complicated to that line. `withSkip (fromJust $ skipDecompress "3:aBc")` feels bad for example. But perhaps an extra `withSkipStr` function would work, which could either error or default to `SkipNothing` if decoding fails. I prefer error.

Reply:
I think it's all good, I can't see it causing any problems, just make sure the compiler output makes the problem obvious if you do give an invalid string. I suspect the `fromJust` would be way worse! haha

Reply:
Well, it's a runtime error, not compile-time. (A compile-time `IsString` would be great.) But I've added `fromString` to the error message to make it a bit clearer where it's coming from.